### PR TITLE
prevent snapshot while onroad

### LINKED ISSUE #2364: Snapshot request when onroad causes openpilot to go offroad
**Describe the bug**

Requesting a snapshot while on onroad causes openpilot to go temporarily offroad. I see the offroad snapshot alert and then eventually dash errors (a bit of delay on Honda Bosch). Onroad does come back after the snapshot sequence finishes.

Note: It looks like the process check in snapshot.py is returning an empty array/list when onroad. Is something killing camerad?

**How to reproduce or log data**

1. Go onroad.
2. Request a snapshot via the API.
3. Device will go offroad and show the snapshot alert. CAN sending appears to stop during the sequence.
4. The snapshots intermittently make it back through via the API. Other times, they don't.

**Expected behavior**

Nothing. Return None and don't pull snapshots when onroad.

**Device/Version information (please complete the following information):**
 - Device: comma two
 - Dongle ID: 95e637a323d6e9fc
 - Routes: These are consecutive. My house requests snapshots every 5 minutes or so, hence the timed nature. 2020-10-19--17-39-48, 2020-10-19--17-43-06, 2020-10-19--17-48-07
 - Timestamp: N/A
 - Version: master-ci 29a96263ea27c7998753a234bcd8c10261ec7fef (10/19/2020)
 - Car make/model: 2017 Civic Hatchback

### PR DESCRIPTION
**Pull request by adeebshihadeh**
fixes #2364

### CODE REVIEW

**Comment by csouers** [comment]
The issue still occurs with this branch. It's as if the `ps | grep camerad` is returning as if camerad is not running.

File: selfdrive/camerad/snapshot/snapshot.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def snapshot():
   params = Params()
   front_camera_allowed = int(params.get("RecordFront"))
 
-  if params.get("IsTakingSnapshot") == b"1":
+  if params.get("IsOffroad") is not None or params.get("IsTakingSnapshot") == b"1":
```
Comment (csouers) [inline selfdrive/camerad/snapshot/snapshot.py:22]: With a change to `  if params.get("IsOffroad") is None or params.get("IsTakingSnapshot") == b"1":
`, this works as expected

Comment (adeebshihadeh) [reply selfdrive/camerad/snapshot/snapshot.py]: oops, should be fixed now

Comment (csouers) [reply selfdrive/camerad/snapshot/snapshot.py]: Tested fine. Thank you!

**Comment by csouers** [comment]
Testing fine now. Thanks!

**Pull request merged**